packages/ketcher-react/src/Editor.tsx

```diff
@@ -64,10 +64,14 @@ function Editor(props: EditorProps) {
       }
     });
     return () => {
-      // setTimeout is used to disable the warn msg from react "Attempted to synchronously unmount a root while React was already rendering"
-      setTimeout(() => {
-        appRoot.unmount();
-      });
+      isUnmounted.current = true;
+      // In StrictMode, React double invokes and the component could be unmounted again due to the timeout.
```

Comment:
A question here is that seems the isUnmounted.current would always be true and the unmount could not be triggered, which also got verified(memory leak still occurs) after I tested it with my old memory-leak-fix branch:
[2912-example-fix-memory-leak-for-ketcher-when-it-is-opened-inside-modal-1](https://github.com/epam/ketcher/tree/2912-example-fix-memory-leak-for-ketcher-when-it-is-opened-inside-modal-1)

Reply:
Great catch! And you're right, it should be below. Unfortunately, this doesn't fix the problem so I investigated further.

This actually exposed the "real" reason why strict mode doesn't work:
Render once, unmount. `setTimeout` is put into the next event loop. 
Second mount. We create the root. We don't really have control when `setTimeout` executes, But from testing  `setTimeout` function is called after the useEffect and `appRoot.unmount();` is executed. This removes the `Editor`.

Reply:
`appRoot.unmount();` should be synchronous so that it executes immediately after the unmount. I think it's worth it even if we get the 
`// setTimeout is used to disable the warn msg from react "Attempted to synchronously unmount a root while React was already rendering"`
. Otherwise folks who use ketcher with strictmode would not be able to use ketcher at all.